WB Attachments: Use bigger image viewer

alesan99 opened the pull request:
Fixes #7111

Switches to Leaflet as the image viewer for attachments uploaded through the workbench.
Also moves dialog buttons to the header for extra viewing space.

Benefits:
- Bigger image viewing area.
- Support for image tiling (for very large images) in the future.

### Checklist

- [x] Self-review the PR after opening it to make sure the changes look good and
      self-explanatory (or properly documented)
- [x] Add relevant issue to release milestone
- [ ] Add pr to documentation list
- [ ] Add automated tests
- [ ] Add a reverse migration if a migration is present in the PR

### Testing instructions
- Import Attachments into the workbench
- Click on any row and open an attachment.
- [ ] Test that the image viewer works as expected. Try zooming, panning, and changing the size of the dialog.
- [ ] Test the reset zoom button
- Click on a different row
- [ ] Make sure the image was updated  in the viewer.
- Click detach window
- [ ] Test that the image viewer works as expected. Try zooming, panning, and changing the size of the window.

alesan99 commented:
> If you have a very large image you cannot zoom out and fit the whole image on the screen even if you detach the image and go full screen on that page.

Leaflet lets me add a custom minimum zoom but I'm going to see if I can also add a custom "fit to screen" button to make it more conventient 👍
I'm also looking into clover-iiif as a potential option https://samvera-labs.github.io/clover-iiif/docs/viewer. It's built specifically for images so I may switch to it if you think that works better

alesan99 commented:
@emenslin Ready for testing again 👌 I made various improvement to zooming, along with a new reset zoom button

emenslin reviewed (approved):
- [x] Test that the image viewer works as expected. Try zooming, panning, and changing the size of the dialog.
- [x] Test the reset zoom button
- [x] Make sure the image was updated  in the viewer.
- [x] Test that the image viewer works as expected. Try zooming, panning, and changing the size of the window.

---

Looks good! The only thing I noticed is that for some smaller images the border around it is large, I think this is fine but I just wanted to point it out and verify that it fits the requirments @grantfitzsimmons 

<img width="766" height="727" alt="image" src="https://github.com/user-attachments/assets/b413dfa0-2b2f-447e-a4ca-29277cc2e35f" />

grantfitzsimmons reviewed (commented):
**Very nice work**, this is really solid and tested with a large number of different file types. 

Detaching the window **doesn't** work in Arc, but works in Chrome and Firefox. Acceptable in my eyes as this is a niche browser.

>Leaflet lets me add a custom minimum zoom but I'm going to see if I can also add a custom "fit to screen" button to make it more conventient 👍

@alesan99 This doesn't seem to work completely consistently, as often the image is not fitting to the frame, but I find it to be acceptable.

One concern I have is with the Leaflet viewer not fitting the size of the dialog making it appear incorrectly.

https://github.com/user-attachments/assets/126c15a3-7e51-49ea-b106-aa39f16530bd

Another is with the icon for reset– can you a hero icon instead? https://heroicons.dev/?iconset=v1-20-solid&search=refresh

<img width="69" height="135" alt="Image" src="https://github.com/user-attachments/assets/87489e04-7c70-4cdd-8852-c730be284dbf" />

alesan99 commented:
<img width="1166" height="873" alt="image" src="https://github.com/user-attachments/assets/300ef04a-62a8-4f79-b8e4-0437d9b6e1d3" />


Added a default dialog size to allow for resizing the dialog to smaller sizes and replaced the reset icon.

Also improved fitting the image to the screen. It no longer leaves a large border around the image.
